pegasus/sites.v3/code.org/public/educate/weblab.html

```diff
@@ -0,0 +1,222 @@
+            action.
+          </p>
+          <p>
+            Contact our Support Team (<a href="mailto:support@code.org"
+              >support@code.org</a
+            >) if you have any questions.
+          </p>
+        </div>
+        <iframe id="test-iframe" src="/" style="display: none"></iframe>
+      </div>
+    </div>
+  </div>
+</div>
+<script>
+  // TODO: Is there a way to use the existing code for this in apps directory?
+  // This is a promisifed version of a function that already exists.
+  const TIMEOUT_MS = 3000;
+  function testImageAccess(url, callback = () => {}) {
+    return new Promise((resolve, reject) => {
+      const element = new Image();
+      element.src = url;
+
+      let succeeded = false;
+      element.onerror = () => {
+        callback(new Error());
+        reject();
+      };
+      element.onload = () => {
+        succeeded = true;
+        callback();
+        resolve();
+      };
+      setTimeout(() => {
+        if (!succeeded) {
+          callback(new Error());
+          reject();
+        }
+      }, TIMEOUT_MS);
+
+      // store a reference to the element so it doesn't get collected
+      window.testImages = window.testImages || [];
+      window.testImages.push(element);
+    });
+  }
+
+  function testIFrameAccess(successCallback, failureCallback) {
+    return new Promise((resolve, reject) => {
+      var iframe = document.getElementById("test-iframe");
+      var iframeDoc = iframe.contentDocument || iframe.contentWindow.document;
+
+      const checkIFrameLoaded = () => {
+        let elapsedTime = new Date().getTime() - startTime;
+        if (iframeDoc.readyState === "complete") {
+          setAttribute(".iframe > .loading", { style: "display: none" });
+          setAttribute(".iframe > .succeeded", {
+            style: "display: block",
+          });
+          resolve();
+        } else if (elapsedTime > TIMEOUT_MS) {
+          setAttribute(".iframe > .loading", { style: "display: none" });
+          setAttribute(".iframe > .failed", {
+            style: "display: block",
+          });
+          reject();
+        }
+      };
+
+      const startTime = new Date().getTime();
+      setInterval(checkIFrameLoaded, 100);
+    });
+  }
+
+  function setAttribute(selector, obj) {
+    Array.from(document.querySelectorAll(selector)).forEach((element) =>
+      Object.keys(obj).forEach((key) => {
+        element.setAttribute(key, obj[key]);
+      })
+    );
+  }
+
+  function connect() {
+    setAttribute(
+      ".incomplete, .succeeded, .failed, #success-message, #fail-message",
+      { style: "display: none" }
+    );
+    setAttribute(".loading", { style: "display: block" });
+
+    const computingInTheCorePromise = testImageAccess(
+      "https://downloads.computinginthecore.org/favicon.ico?" + Math.random(),
+      (err) => {
+        if (!err) {
+          setAttribute(".computinginthecore > .loading", {
```

Comment:
We should build the UI in React.  This will be fairly easy once building in apps/.